src/addon.cpp

```diff
@@ -66,35 +66,39 @@ class SHA3Hash: public ObjectWrap {
 
 	static
 	NAN_MODULE_INIT(Init) {
-		Local<FunctionTemplate> t = Nan::New<FunctionTemplate>(New);
-		t->SetClassName(Nan::New<String>("SHA3Hash").ToLocalChecked());
-		t->InstanceTemplate()->SetInternalFieldCount(1);
+		Local<Context> context = target->CreationContext();
 
-		Nan::SetPrototypeMethod(t, "update", Update);
-		Nan::SetPrototypeMethod(t, "digest", Digest);
+		Local<String> className = Nan::New<String>("SHA3Hash").ToLocalChecked();
 
-		constructor.Reset(t->GetFunction());
-		target->Set(Nan::New<String>("SHA3Hash").ToLocalChecked(), t->GetFunction());
+		Local<FunctionTemplate> functionTemplate = Nan::New<FunctionTemplate>(New);
+
+		functionTemplate->SetClassName(className);
+		functionTemplate->InstanceTemplate()->SetInternalFieldCount(1);
+
+		Nan::SetPrototypeMethod(functionTemplate, "update", Update);
+		Nan::SetPrototypeMethod(functionTemplate, "digest", Digest);
+
+		Local<Function> f = functionTemplate->GetFunction(context).ToLocalChecked();
+		constructor.Reset(f);
+		target->Set(className, f);
 	}
 
 	static
 	NAN_METHOD(Update) {
+		Isolate *isolate = info.GetIsolate();
+		Local<Context> context = isolate->GetCurrentContext();
 		SHA3Hash *obj = ObjectWrap::Unwrap<SHA3Hash>(info.This());
 
 		ASSERT_IS_STRING_OR_BUFFER(info[0]);
-#if NODE_VERSION_AT_LEAST(0,11,0)
 		enum Nan::Encoding enc = static_cast<Nan::Encoding>(ParseEncoding(Isolate::GetCurrent(), info[0], node::BINARY));
-#else
-		enum Nan::Encoding enc = static_cast<Nan::Encoding>(ParseEncoding(info[0], node::BINARY));
-#endif
```

Comment:
:memo: **Implementation Note:** This (and the other conditional on L122) has been removed because these very old versions of Node.js (< v0.11.0) are no longer supported.